### ISSUE TITLE
AttributeError: __exit__ on "with" statement

### ISSUE THREAD

**Issue by vfilimonov** [joblib/joblib#241]
An example from https://pythonhosted.org/joblib/parallel.html#reusing-a-pool-of-workers:
```
with Parallel(n_jobs=2) as parallel:
    accumulator = 0.
    n_iter = 0
    while accumulator < 1000:
        results = parallel(delayed(math.sqrt)(accumulator + i ** 2)
                           for i in range(5))
        accumulator += sum(results)  # synchronization barrier
        n_iter += 1
```
raises an Exception `AttributeError: __exit__`.
But the direct code:
```
parallel = Parallel(n_jobs=2)

accumulator = 0.
n_iter = 0
while accumulator < 1000:
    results = parallel(delayed(math.sqrt)(accumulator + i ** 2)
                       for i in range(5))
    accumulator += sum(results)  # synchronization barrier
    n_iter += 1
```
works fine.

Joblib version 0.8.4.

**Comment by vfilimonov** [comment]
Ah! eventually it was fixed in 0.9 beta.

**Comment by GaelVaroquaux** [comment]
> Ah! eventually it was fixed in 0.9 beta.

This syntax was added in version 0.9.

Note to self: we should start using the sphinx primitive 'versionadded'
to note this in the documentation.
